src/web/lib/components/SavedThemeSelector/index.js

```diff
@@ -8,21 +8,30 @@ export const SavedThemeSelector = ({
   savedThemesPage,
   setSavedThemesPage,
   themeCustomImages,
-  storage
+  storage,
+  clearCustomBackground,
+  setCurrentThemeId
 }) => {
   const { themeStorage } = storage;
   const sortedSavedThemes = Object.entries(savedThemes).sort(
     ([, aData], [, bData]) => bData.modified - aData.modified
   );
 
+  const deleteTheme = (key) => {
+    // This is so we can track which theme to delete in some other places.
+    setCurrentThemeId({ key });
```

Comment:
Why is this needed? I expect that `selectors.savedThemes(state)` already returns themes without the deleted theme, simply because it is not in the storage any more. Is the theme somehow returning without this hack?

When `themeStorage.delete` is called, the "storage" event is triggered, which updates the saved themes already: https://github.com/mozilla/FirefoxColor/blob/17092506ea355f49894eafe406a675cc114af9f3/src/web/lib/storage.js#L147-L150

One thing that seems to be missing is that `currentSavedTheme` is still at the old value. I suppose that it may be necessary to call `this.afterPut(undefined);` to properly dispose of the cleared theme in `delete` at
https://github.com/mozilla/FirefoxColor/blob/17092506ea355f49894eafe406a675cc114af9f3/src/web/lib/storage.js#L55

Reply:
hey, thanks again for the feedback :)

This is a bit trickier than I thought but I'll try to take a look over the next few days.. But if I recall, the theme was still in there which is why I approached it this way. I'll take another look at using afterPut. maybe that's the key..? I'll report back :)